File: 20221013/모범답안/20221013_03/keys.js
/**
 * @param {str} keys
 * @param {str} doors
 * @return {str}
 */
function solution(keys, doors) {
    keys = keys.split('');
    doors = doors.split('');

    while (keys.length) {
        if (keys.pop().toUpperCase() !== doors.shift()) {
            return 'NO'
        }
    }
    
    return 'YES'
}

File: 20221012/모범답안/20221012_03/keys.py
def solution(keys, doors):
  '''
  :param keys: str
  :param doors: str
  :return: str
  '''
  keys = list(keys)
  doors = list(doors)
  
  while keys:
    if keys.pop(-1).upper() != doors.pop(0):
      return 'NO'
  
  return 'YES'
